Issue #194 Update README to improve installation guide.

bendiy opened the pull request:
For: https://github.com/plv8/plv8/issues/194

bendiy commented:
TODO: https://github.com/plv8/plv8/issues/161

JerrySievert commented:
thanks for your work here, but ultimately I'd like to see this information go into `plv8.md` under `doc`, as this gets generated into `PGXN` documentation as well:

http://pgxn.org/dist/plv8/doc/plv8.html

will reply inline as to what might work well with README.md, and what might work better with plv8.md.

JerrySievert commented on README.md:
```diff
@@ -1,132 +1,212 @@
+
 - PG: version 9.2, 9.3, 9.4 and 9.5 (maybe older/newer are allowed)
-- V8: version 4.4 to 4.10
+- V8: version 4.4 to 4.10
```
actually, up to 5.1 at this point.

JerrySievert commented on README.md:
````diff
@@ -1,132 +1,212 @@
+```shell
+$ git clone git@github.com:plv8/plv8.git
+$ cd ./plv8
+$ git checkout tags/v1.5.4 -b 1.5.4
````
I would rather not get people checking out branches unless they have very special requirements.  for instance, `1.4` is maintained strictly for Debian compatibility, as they work their way to a modern `v8`.  `1.5` is maintained for legacy at this point as well.

great pains are made to not break master, where development on modern `v8` is being done, and more modern OS's will either ship with a newer `libv8`, or will require a more recent version of `v8` in order to compile at all without hacks.

ruimarinho replied:
@JerrySievert wdyt about releasing current master HEAD as 2.0.0-dev1? That way I could probably tag a docker postgres-plv8 version for more widespread usage.

JerrySievert commented on README.md:
````diff
@@ -1,132 +1,212 @@
+$ make static
+# Then move it back.
+$ mv /usr/lib/9.6 /usr/lib/postgresql/
+```
````
I feel this whole section should exist in `plv8.md` - a link would suffice.

bendiy commented:
@JerrySievert Thanks for the feedback. I'll make a few more tweaks to this next week.

bendiy commented on README.md:
```diff
@@ -1,132 +1,210 @@
+function that is callable from SQL.
+
+# Requirements:
+plv8 is tested with:
```
@JerrySievert Do you know if the:
```
REQUIREMENT
-----------
```

Format is needed for the PGXN wiki or if I can use the Markdown `#` formatter for titles?
```
# Requirements:
```

bendiy commented:
@JerrySievert I was able to get back to this PR and finish it. I've moved pretty much all of the README to the doc/plv8.md file and cleaned up the formatting.

JerrySievert commented:
will review soon, but in response to @ruimarinho's now missing (GitHub!!!!!!) comment, I've been thinking about it for a while, and while I'd love to release 2.0 with a debugger, it might be best to wait for 2.1 with that - I'll work on doing some cleanup and getting 2.0 released soon.

bendiy commented:
Missing comment is here:
https://github.com/plv8/plv8/pull/198#discussion_r96537283

JerrySievert commented:
thanks @bendiy :) also just asked for a feature request to get a link to the branch instead of just a hover over ...

JerrySievert commented:
@bendiy I'll give some notes inline, but after that, considering @ruimarinho's request, I'd love it if you'd put in a PR against the 1.5 branch as well.

JerrySievert commented:
(otherwise, this is awesome)

JerrySievert commented on doc/plv8.md:
```diff
@@ -1,50 +1,248 @@
+### Redhat/CentOS:
+TODO - PL/v8 supports Redhat/CentOS. A Pull Request for installation steps is greatly appreciated.
+
+### MacOS:
```
```
$ brew install plv8
```

or `make static` and `make install`

bendiy replied:
done

JerrySievert commented on doc/plv8.md:
````diff
@@ -1,50 +1,248 @@
+    plv8.elog(WARNING, 'plv8.version = ' + plv8.version); -- Will output the PL/v8 installed as a PostgreSQL `WARNING`.
+  $$ LANGUAGE plv8;
+```
+As of 2.0.0, there is a function to determine which version of PL/v8 you have
````
I really need to backport this (note for myself, not for you)

bendiy commented:
@JerrySievert 
I can add a PR for the 1.5 branch. I'll do that once you approve and merge.

JerrySievert commented:
merged - I'll do an update on checkout when I cut the 2.0 release (likely this weekend)

JerrySievert commented:
btw, awesome, thanks much!

bendiy commented:
No problem. Thanks for all your work on this project too.
`r1.5` branch PR is here:
https://github.com/plv8/plv8/pull/204
